qiskit_nature/properties/__init__.py

```diff
@@ -33,6 +33,17 @@
 
 from .grouped_property import GroupedProperty
 from .property import Property
+from ..deprecation import warn_deprecated, DeprecatedType, NatureDeprecationWarning
+
+warn_deprecated(
+    "0.5.0",
+    old_type=DeprecatedType.PACKAGE,
+    old_name="qiskit_nature.properties",
+    new_type=DeprecatedType.PACKAGE,
+    new_name="qiskit_nature.second_q.properties",
```

Comment:
This is not entirely true. Some of the properties will end up in `qiskit_nature.second_q.hamiltonians`.
Some other utilities in here (e.g. the `Integrals` classes) will be replaced by the `PolynomialTensor`.

I think this requires a bit more granular work unfortunately.

Reply:
Where do the properties go ? Should we wait until classes like  `PolynomialTensor` are merged so that we add proper deprecation messages ?

Reply:
Here is an overview of the classes in properties and where they will end up:
```
GroupedProperty -> removed, no replacement necessary
Property -> refactored, base class as this no longer existed in the same form (if at all, it will be `qiskit_nature.second_q.properties.Property`)

DriverMetadata -> qiskit_nature.second_q.problems.DriverMetadata (maybe rename to `Precedence` using naming from QCSchema)

AngularMomentum -> qiskit_nature.second_q.properties
Magnetization -> qiskit_nature.second_q.properties
ParticleNumber -> qiskit_nature.second_q.properties
(Electronic)DipoleMoment -> qiskit_nature.second_q.properties
ElectronicEnergy -> qiskit_nature.second_q.hamiltonians
ElectronicIntegrals -> succeeded by qiskit_nature.second_q.operators.PolynomialTensor
OneBodyElectronicIntegrals -> PolynomialTensor
TwoBodyElectronicIntegrals -> PolynomialTensor
IntegralProperty -> refactored, no longer needed as such
ElectronicStructureDriverResult -> succeeded by ElectronicStructureProblem

OccupiedModals -> qiskit_nature.second_q.properties
VibrationalEnergy -> qiskit_nature.second_q.hamiltonians
VibrationalIntegrals -> succeeded by qiskit_nature.second_q.operators.PolynomialTensor
VibrationalStructureDriverResult -> succeeded by VibrationalStructureProblem

ElectronicBasis -> TBD
VibrationalBasis -> TBD
```